recipes/ruby/all/test_package/conanfile.py

```diff
@@ -1,21 +1,71 @@
-from conan import ConanFile
-from conan.tools.cmake import CMake
-from conans import tools
 import os
+import re
+from io import StringIO
+
+from conan import ConanFile
+from conan.tools.build import can_run
+from conan.tools.cmake import CMake, CMakeToolchain, cmake_layout
+
 
 class TestPackageConan(ConanFile):
-    settings = "os", "compiler", "build_type", "arch"
-    generators = "CMakeDeps", "CMakeToolchain"
+    settings = "os", "arch", "compiler", "build_type"
+    generators = "CMakeDeps", "VirtualRunEnv"
+
+    def requirements(self):
+        self.requires(self.tested_reference_str)
+
+    def build_requirements(self):
+        self.tool_requires(self.tested_reference_str)
```

Comment:
You should be able to get rid of the "missing binary" error in CI by avoiding a `tool_requires()` on the tested package.
```diff
@@ -14,9 +14,6 @@
     def requirements(self):
         self.requires(self.tested_reference_str)
 
-    def build_requirements(self):
-        self.tool_requires(self.tested_reference_str)
-
     def layout(self):
         cmake_layout(self)
 
@@ -44,7 +41,7 @@
     def _test_ruby_executable_version(self):
         # test executable
         output = StringIO()
-        self.run("ruby --version", output, env="conanbuild")
+        self.run("ruby --version", output, env="conanrun")
         output_str = str(output.getvalue()).strip()
         self.output.info(f"Installed version: {output_str}")
         assert_ruby_version = f"ruby {self._ruby_version()}"
@@ -54,7 +51,7 @@
     def _test_ruby_execute(self):
         # test executable
         output = StringIO()
-        self.run('ruby -e "puts RUBY_VERSION"', output, env="conanbuild")
+        self.run('ruby -e "puts RUBY_VERSION"', output, env="conanrun")
         output_str = str(output.getvalue()).strip()
         self.output.info(f'ruby -e "puts RUBY_VERSION": "{output_str}"')
         assert_ruby_version = self._ruby_version()
```

Reply:
I had been advised to use a tool_requires. I've just tried removing it, let's see what CI says...
I did have to use `self.runenv_info.prepend_path("PATH", os.path.join(self.package_folder, "bin"))`